### PR TITLE
Handle authentication (pt. 2)

### PR DESCRIPTION
**Pull request by Tom-Bonnike**
In this PR, we try to automatically log users out when they make a request with an invalid/obsolete token to the API.
From my manual tests, I believe it should work but we might have to tweak some stuff when we actually make queries :)

### CODE REVIEW

File: src/helpers/clearAccessToken.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const clearAccessToken = (context: any): void => {
+  const epochTimestamp = Math.floor(new Date() / 1000)
+  const cookies = serializeCookie(AUTH_COOKIE, '', {
+    // Setting a `maxAge` in the past expires the cookie.
+    maxAge: epochTimestamp
+  })
+
+  if (context && context.ctx.res) {
+    const response = context.ctx.res
+    response.setHeader('Set-Cookie', cookies)
```
Comment (Tom-Bonnike) [inline src/helpers/clearAccessToken.js]: We don’t actually clear the token server-side but I guess we could want to do that at some point

File: src/higherOrders/withApolloClient/initApollo.js [hunk 2]
```diff
@@ -3,35 +3,46 @@
...
+const customFetch = (context: any) => async (URI: string, options: any) => {
   const token = getAccessToken()
   options.headers.Authorization = token ? `bearer ${token}` : ''
 
-  return fetch(URI, options)
+  const response = await fetch(URI, options)
+
+  // We intercept the response and log the user out if their access token is not
+  // valid.
+  if (response.status === 401) {
```
Comment (Tom-Bonnike) [inline src/higherOrders/withApolloClient/initApollo.js]: I don’t think we throw a 401 for other reasons and we probably shouldn’t :)

File: src/higherOrders/withApolloClient/initApollo.js [hunk 3]
```diff
@@ -3,35 +3,46 @@
...
 import fetch from 'isomorphic-unfetch'
 import getAccessToken from '@helpers/getAccessToken'
+import clearAccessToken from '@helpers/clearAccessToken'
+import redirectTo from '@helpers/redirectTo'
 
 let apolloClient = null
 
-const customFetch = (URI, options) => {
+const customFetch = (context: any) => async (URI: string, options: any) => {
   const token = getAccessToken()
```
Comment (Tom-Bonnike) [inline src/higherOrders/withApolloClient/initApollo.js]: we should pass `context` here.

File: src/higherOrders/withApolloClient/initApollo.js (hunk 2 above)
Comment (rmnbrd) [reply src/higherOrders/withApolloClient/initApollo.js]: 👍

**Pull request merged**